### ISSUE TITLE
OSX specific: wxLocale: Problem with set locale for any of wxLANGUAGE_CHINESE(...)

### ISSUE THREAD

**Issue by YuSanka** [wxWidgets/wxWidgets#23209]
### Description

When we try to switch our application to the some of Chinese languages, then wxWidgets send a message, that cannot set locale for this language.

<img width="279" alt="Screenshot 2023-02-02 at 16 29 04" src="https://user-images.githubusercontent.com/19184891/216372164-7530bf74-ff53-4f15-9e59-de07de6d7021.png">

The bug appears only on macOS

### Platform and version information

- wxWidgets version  you use: 3.2.0
- wxWidgets port  you use: wxOSX
- OS  and its version: tested on Monterey and Ventura

**Comment by vadz** [comment]
Could you please explain what exactly are you doing, i.e. how are you setting the locale? Some code, or, ideally, a diff to the internat sample would be really welcome.

This would at least make it clear if this is related to #23114 or something different.

**Comment by vadz** [comment]
Also, please check if the changes in #23226 change anything for the problem you're seeing.

**Comment by YuSanka** [comment]
Hello, @vadz . Sorry for the late response. But we have to reopen this issue because of it's not fix yet. 
I tried all motioned issues/fixes but, behavior is the same.
To simplest repro just change a wxLANGUAGE_DEFAULT to wxLANGUAGE_CHINESE_TAIWAN for "calendar" sample of wxWidgets.

<img width="552" alt="Screenshot 2023-05-18 at 14 08 21" src="https://github.com/wxWidgets/wxWidgets/assets/19184891/70d18d90-16e7-4ff6-9315-c39a1e3ddf3f">

Run "calendar" application and you will receive a warning which I described.

<img width="1074" alt="Screenshot 2023-05-18 at 14 04 12" src="https://github.com/wxWidgets/wxWidgets/assets/19184891/556b5961-0338-4cd7-b2b0-9aa8693084b1">

**Comment by vadz** [comment]
OK, I can indeed reproduce the problem with a very simple test case:
```cpp
#include <wx/init.h>
#include <wx/crt.h>
#include <wx/datetime.h>
#include <wx/intl.h>

int main(int argc, char **argv)
{
    wxInitializer init;
    if ( !init.IsOk() ) {
        fprintf(stderr, "Failed to initialize wxWidgets.\n");
        return 3;
    }

    wxPrintf("Without locale: %s\n", wxDateTime::Today().FormatDate());
    wxLocale loc(wxLANGUAGE_CHINESE_TAIWAN, wxLOCALE_DONT_LOAD_DEFAULT);
    if ( !loc.IsOk() )
        wxPrintf("Setting locale failed.\n");
    else
        wxPrintf("With locale:    %s\n", wxDateTime::Today().FormatDate());

    return 0;
}
```

The reason it happens is that `[NSLocale availableLocaleIdentifiers]` doesn't contain `zh_TW`, but only `zh_Hant_TW`, so we fail to initialize this locale even if it is, in fact, supported by the system.

I've fixed this in #23556 and your original example should work too now, but, as described in the commit message there, I'm not sure if it works correctly for `zh_CN` locale which exists in both `zh_Hans_CN` (which you'd normally want) and `zh_Hant_CN` versions (I know that you could use `wxLANGUAGE_CHINESE_SIMPLIFIED_EXPLICIT` or `wxLANGUAGE_CHINESE_TRADITIONAL_EXPLICIT` to select them explicitly). Any thoughts from people actually using/reading Chinese and using macOS would be welcome, but for now this should work at least somehow.

Please let me know if you still see any problems with that PR (and **please** provide text diffs/logs instead of screenshots if you do).

**Comment by YuSanka** [comment]
Hello, @vadz. I can confirm, that this commit fixed our issue.
Thank you very much
